lm_eval/utils.py

```diff
@@ -84,13 +85,19 @@ def complete_code(
         with torch.no_grad():
             if task.stop_words:
                 gen_kwargs["stopping_criteria"][0].start_length = batch["ids"].shape[-1]
+
+            if batch["ids"].shape[0]==1:
+                batch["ids"] = batch["ids"][:,:batch["input_len"]]
+                batch["attention_mask"] = batch["attention_mask"][:,:batch["input_len"]]
```

Comment:
IIURC you only remove padding here if there is a single sample in the input i.e. batch size is 1, correct? This means that in other cases, the generation would continue with the prior token possibly being a padding token no?

I guess this is why you set `tokenizer.padding_side = "left"` in the test case, but we should also make it work with tokenizers where the padding side is right I think.

Reply:
Thanks for the review.  

Yes, The padding is removed when batch_size is 1. For the other case (batch_size > 1), We are setting the `padding_side="left"` , as suggested in [this issue](https://github.com/huggingface/transformers/issues/3021#issuecomment-1455920110) to handle the influence of padding token in the generation when `padding_side="right"`( also as done in the [fix](https://github.com/huggingface/transformers/pull/7552/files#diff-d960b97297020dd1a0416977d73f40dee77ef969df0848a981b7f46c2ca6373cR415) to this issue).